packages/@react-stately/steplist/src/useStepListState.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaStepListProps, StepListState} from '@react-types/steplist';
+import {Collection, Node} from '@react-types/shared';
+import {Key, useMemo} from 'react';
+import {useControlledState} from '@react-stately/utils';
+import {useSingleSelectListState} from '@react-stately/list';
+
+export function useStepListState<T extends object>(props: AriaStepListProps<T>): StepListState<T> {
+  let selectState = useSingleSelectListState<T>(props);
+  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key>(props.lastCompletedStep, props.defaultCompletedStep, props.onLastCompletedStepChange);
+  const collection = selectState.collection as Collection<Node<T>>;
+  const keys = useMemo<Key[]>(() => iterToArr(collection.getKeys()), [collection]);
+  const {setSelectedKey} = selectState;
+  if (keys.indexOf(selectState.selectedKey) > keys.indexOf(lastCompletedStep)) {
+    setLastCompletedStep(selectState.selectedKey);
+  }
+  const sls: StepListState<T> = {
```

Comment:
can we move this to the bottom? i spent a few min trying to find a few of these properties only to realize they were below